core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -153,6 +155,18 @@ public WriteBuilder metricsConfig(MetricsConfig newMetricsConfig) {
       return this;
     }
 
+    /**
+     * Overwrite only if needed on File systems where a new empty file gets pre-created like {@link
+     * HadoopFileIO}
+     *
+     * @return {@link WriteBuilder}
+     */
+    public WriteBuilder overwriteIfNeeded() {
+      boolean needsOverwrite =
+          file instanceof HadoopFileIO || file instanceof Files.LocalOutputFile;
```

Comment:
does `LocalOutputFile` also gets pre-created via new empty file? Do we really need to make this class public in order to do this?

Reply:
yes its used in unit tests and `tmp.newFile` pre-creates a local empty file.

Had to make it public to be used in this class here, as package is different I couldn't use protected/package-protected

Reply:
I'm probably missing something here but file is a `OutputFile` can it ever be an instanceof `HadoopFileIO`? That seems possible if a class implements both interfaces which doesn't make sense to me. Also if it's based on some notion of OutputFile or FileIO have we considered a mixin interface instead of adding to the API?